kite-morphlines/kite-morphlines-core/src/main/java/org/kitesdk/morphline/stdlib/HashDigestBuilder.java

```diff
@@ -0,0 +1,215 @@
+
+/**
+ * Command that creates a hash digest of a field so-as to deterministically
+ * provide a unique key
+ */
+
+public class HashDigestBuilder implements CommandBuilder {
+
+  @Override
+  public Collection<String> getNames() {
+    return Collections.singletonList("hashDigest");
+  }
+
+  @Override
+  public Command build(Config config, Command parent, Command child,
+      MorphlineContext context) {
+    return new HashDigest(this, config, parent, child, context);
+  }
+
+  ///////////////////////////////////////////////////////////////////////////////
+  // Nested classes:
+  ///////////////////////////////////////////////////////////////////////////////
+  private static final class HashDigest extends AbstractCommand {
+
+    private final String inputFieldName;
+    private final String outputFieldName;
+    private final String hashType;
+    private final boolean preserveExisting;
+
+    private static final String INPUT_FIELD = "inputField";
+    private static final String OUTPUT_FIELD = "outputField";
+    private static final String HASH_TYPE = "hashType";
+    private static final String PRESERVE_EXISTING_NAME = "preserveExisting";
+    private static final boolean PRESERVE_EXISTING_DEFAULT = true;
+
+    public HashDigest(CommandBuilder builder, Config config, Command parent,
+        Command child, MorphlineContext context) {
+      super(builder, config, parent, child, context);
+
+      this.inputFieldName = getConfigs().getString(config, INPUT_FIELD);
+      this.outputFieldName = getConfigs().getString(config, OUTPUT_FIELD);
+      this.hashType = getConfigs().getString(config, HASH_TYPE);
+      this.preserveExisting = getConfigs().getBoolean(config,
+          PRESERVE_EXISTING_NAME, PRESERVE_EXISTING_DEFAULT);
+
+      validateArguments();
+
+      if (LOG.isTraceEnabled()) {
+        LOG.trace("inputField: {}", inputFieldName);
+        LOG.trace("outputField: {}", outputFieldName);
+        LOG.trace("hashType: {}", hashType);
+        LOG.trace("preserveExisting: {}", preserveExisting );
+      }
+    }
+
+    @Override
+    protected boolean doProcess(Record record) {
+
+      if (!preserveExisting) {
+        record.removeAll(outputFieldName);
+      }
+
+      ListIterator iter = record.get(inputFieldName).listIterator();
+      
+      while (iter.hasNext()) {
+        record.put(outputFieldName, doHash(iter.next().toString()));
+      }
+
+      // pass record to next command in chain:
+      return super.doProcess(record);
+
+    }
+
+    private String doHash(String inputString) {
+      MessageDigest digest = DigestUtils.getDigest(hashType);
+
+      byte[] inputBytes = StringUtils.getBytesUtf8(inputString);
+
+      return Hex.encodeHexString(digest.digest(inputBytes));
+    }
+
+    @Override
+    protected void doNotify(Record notification) {
+      super.doNotify(notification);
+    }
+
+  }
+  
+  
+  //Manually shaded from org.apache.commons.codec 1.10.
+  public static class DigestUtils {
```

Comment:
I think you can just delete the entire DigestUtils class and simply paste the getDigest(hashType) one liner into your doHash() method. This removes unnecessary dead code and improves maintainability.